File: interpreter-for-cpp/src/Runtime/Value/ValueNull.cpp
#include "ValueNull.h"

using namespace peak::interpreter;

const std::shared_ptr<ValueNull> ValueNull::DEFAULT_VALUE = std::shared_ptr<ValueNull>(new ValueNull());

std::string ValueNull::ToString() const {
	return "null";
}
std::shared_ptr<Value> ValueNull::Clone() const {
	return std::shared_ptr<Value>(new ValueNull());
}

File: interpreter-for-cs/src/Runtime/Value/ValueNull.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {
	public class ValueNull : TypeValue<ValueNull> {
		public static readonly ValueNull DEFAULT_VALUE = new ValueNull();
		public override string ToString() {
			return "null";
		}
		public override Value Clone() {
			return DEFAULT_VALUE;
		}
	}
} // namespace peak.interpreter